File: onlineTest-2/src/main/java/com/capg/onlinetest/service/TestService.java
package com.capg.onlinetest.service;

import java.util.List;
import com.capg.onlinetest.entity.Test;

/**
 * 
 * @author TEAM EINSTEIN
 * TEST SERVICE INTERFACE
 */
public interface TestService {

	public String addTest(Test test);
    public String deleteTest(int testId);
    public String updateTest(int testId, Test test);
    public List<Test> viewAllTests();
    public float calculateTotalMarks(Test test);
}

File: OnlineTestMgtSystem/src/app/service/test.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";



@Injectable({
  providedIn: 'root'
})
export class TestService {

  constructor(private httpClient: HttpClient) { }

  public viewAllTest() {
    return this.httpClient.get("http://localhost:9090/admin/viewAllTests");
  }
  public addTest(Test){
    return this.httpClient.post("http://localhost:9090/admin/addTest", Test, {responseType:'text' as 'json'});
  }
  public updateTest(Test){
    return this.httpClient.put("http://localhost:9090/admin/updateTest", Test);
  }
  public deleteTest(testId){
    return this.httpClient.delete("http://localhost:9090/admin/deleteTest/" + testId);
  }
}
